src/core/services/gesture/gesture.js

```diff
@@ -81,16 +81,55 @@
     };
 
     if (self.isHijackingClicks) {
+      var maxClickDistance = 6;
       self.handler('click', {
         options: {
-          maxDistance: 6
+          maxDistance: maxClickDistance
         },
-        onEnd: function (ev, pointer) {
+        onEnd: checkDistanceAndEmit('click')
+      });
+
+      self.handler('focus', {
+        options: {
+          maxDistance: maxClickDistance
+        },
+        onEnd: function(ev, pointer) {
           if (pointer.distance < this.state.options.maxDistance) {
-            this.dispatchEvent(ev, 'click');
+            if (canFocus(ev.target)) {
+              this.dispatchEvent(ev, 'focus', pointer);
+              ev.target.focus();
+            }
+          }
+
+          function canFocus(element) {
+            return element.hasAttribute('href') ||
+              (( element.nodeName == 'INPUT' || element.nodeName == 'BUTTON' || 
+                element.nodeName == 'SELECT' || element.nodeName == 'TEXTAREA' ) && !element.hasAttribute('DISABLED')) ||
+              ( element.hasAttribute('tabindex') && element.getAttribute('tabindex') != '-1' );
```

Comment:
* `tabindex="-1"` makes even things like `input` and `button` not focus-able, so that should be at the beginning. 
* Disabled elements can't be focused
* `video` and `audio` and focusable by default.